system-addon/lib/ManualMigration.jsm

```diff
@@ -18,7 +17,22 @@ XPCOMUtils.defineLazyModuleGetter(this, "ProfileAge", "resource://gre/modules/Pr
 this.ManualMigration = class ManualMigration {
   constructor() {
     Services.obs.addObserver(this, MIGRATION_ENDED_EVENT);
-    this._prefs = new Prefs();
+  }
+
+  get getMigrationLastShownDate() {
+    return this.store.getState().Prefs.values.migrationLastShownDate;
+  }
+
+  set setMigrationLastShownDate(newDate) {
+    this.store.getState().Prefs.values.migrationLastShownDate = newDate;
+  }
+
+  get getMigrationRemainingDays() {
+    return this.store.getState().Prefs.values.migrationRemainingDays;
+  }
+
+  set setMigrationRemainingDays(newDate) {
+    this.store.getState().Prefs.values.migrationRemainingDays = newDate;
```

Comment:
We don't want to directly modify the store without notifying others. See `expireMigration` in this file.